fe/fe-core/src/main/java/org/apache/doris/planner/OlapTableSink.java

```diff
@@ -212,7 +212,7 @@ private TOlapTableSchemaParam createSchema(long dbId, OlapTable table) {
             List<String> columns = Lists.newArrayList();
             List<TColumn> columnsDesc = Lists.newArrayList();
             List<TOlapTableIndex> indexDesc = Lists.newArrayList();
-            columns.addAll(indexMeta.getSchema().stream().map(Column::getName).collect(Collectors.toList()));
+            columns.addAll(indexMeta.getSchema().stream().map(Column::getNonShadowName).collect(Collectors.toList()));
```

Comment:
The column with `__doris_shadow_` prefix can only be created when doing `modify column`.
And most of `modify column` operation is not a "light schema change" operation.
But there is an exception, which is doing "modify varchar column's length", which is a `modify column` operation
and will do "light schema change".

So I think we should handle it in `SchemaChangeHandler.processModifyColumn()`, to avoid creating column with
shadow prefix if it is a light schema change, instead of modifying here.

And on BE side, there will still be column with shadow prefix, eg, modify column without light schema change

Reply:
Oh, I see.
The `columns` here is for loading task, and the schema will be saved in rowset's meta(not tablet's meta).
And for rowset meta, there should be no column with `shadow` prefix.